### ISSUE TITLE
PhysicsManager crash

### ISSUE THREAD

**Issue by WickedSchnitzel** [anegostudios/VintageStory-Issues#4328]
### Game Version

1.20-pre6

### Platform

Windows

### Modded

Modded

### SP/MP

Singleplayer

### Description

Singleplayer shut down because of too many errors. Was just flying around, testing schematic placement.

### Logs

```
19.10.2024 13:20:44 [Error] Exception: Object reference not set to an instance of an object.
   at Vintagestory.Server.PhysicsManager.DoWork() in VintagestoryLib\Server\PhysicsManager.cs:line 578
   at Vintagestory.Common.LoadBalancer.SynchroniseWorkToMainThread() in VintagestoryLib\Common\LoadBalancer.cs:line 81
   at Vintagestory.Server.PhysicsManager.DoServerTick() in VintagestoryLib\Server\PhysicsManager.cs:line 574
   at Vintagestory.Server.PhysicsManager.ServerTick(Single dt) in VintagestoryLib\Server\PhysicsManager.cs:line 466
   at Vintagestory.Common.GameTickListener.OnTriggered(Int64 ellapsedMilliseconds) in VintagestoryLib\Common\Model\GameTickListener.cs:line 27
   at Vintagestory.Common.EventManager.TriggerGameTick(Int64 ellapsedMilliseconds, IWorldAccessor world) in VintagestoryLib\Common\EventManager.cs:line 57
   at Vintagestory.Server.CoreServerEventManager.TriggerGameTick(Int64 ellapsedMilliseconds, IWorldAccessor world) in VintagestoryLib\Server\ServerEventManager.cs:line 123
   at Vintagestory.Server.ServerMain.Process() in VintagestoryLib\Server\ServerMain.cs:line 904
```

**Comment by WickedSchnitzel** [comment]
This may be word gen mod related. Seems like it does not occur when https://mods.vintagestory.at/vanillaplus is not installed.

**Comment by WickedSchnitzel** [comment]
also happens in pre7

```[Error] Exception: Object reference not set to an instance of an object.
   at Vintagestory.Server.PhysicsManager.DoWork() in VintagestoryLib\Server\PhysicsManager.cs:line 578
   at Vintagestory.Common.LoadBalancer.SynchroniseWorkToMainThread() in VintagestoryLib\Common\LoadBalancer.cs:line 81
   at Vintagestory.Server.PhysicsManager.DoServerTick() in VintagestoryLib\Server\PhysicsManager.cs:line 574
   at Vintagestory.Server.PhysicsManager.ServerTick(Single dt) in VintagestoryLib\Server\PhysicsManager.cs:line 466
   at Vintagestory.Common.GameTickListener.OnTriggered(Int64 ellapsedMilliseconds) in VintagestoryLib\Common\Model\GameTickListener.cs:line 27
   at Vintagestory.Common.EventManager.TriggerGameTick(Int64 ellapsedMilliseconds, IWorldAccessor world) in VintagestoryLib\Common\EventManager.cs:line 57
   at Vintagestory.Server.CoreServerEventManager.TriggerGameTick(Int64 ellapsedMilliseconds, IWorldAccessor world) in VintagestoryLib\Server\ServerEventManager.cs:line 123
   at Vintagestory.Server.ServerMain.Process() in VintagestoryLib\Server\ServerMain.cs:line 904```

**Comment by WickedSchnitzel** [comment]
Could somebody give a hint if this is a vanilla problem or something within my own mod (server pack)? And what could the problem be/ what type of file or code would i have to edit? It worked without problems before 1.20

**Comment by maltiez2** [comment]
> Could somebody give a hint if this is a vanilla problem or something within my own mod (server pack)? And what could the problem be/ what type of file or code would i have to edit? It worked without problems before 1.20

Better to ask in mods-general channel on official VS discord server

**Comment by nimizin** [comment]
Confirmed still happening in 1.20 pre13, brand new Windows 10 install, no previous VS install. Logs attached.
[server-main.zip](https://github.com/user-attachments/files/17894030/server-main.zip)

**Comment by radfast** [comment]
This issue was fixed in 1.20-rc.1 (by temporarily limiting MaxPhysicsThreads to 1 at loading time).  It is fixed properly in upcoming 1.20-rc.2 where multithreaded physics will be effective again.

Any server still running 1.20-pre versions (not recommended!) instead of 1.20-rc.1, can fix this manually by changing `MaxPhysicsThreads` value to 1 in `servermagicnumbers.json`
